addons/point_of_sale/models/pos_session.py

```diff
@@ -300,15 +305,485 @@ def _validate_session(self):
+            'name': '%s - %s' % (self.name, payment.payment_method_id.name),
+        }
+        return self._debit_amounts(partial_vals, amount, amount_converted)
+
+    def _get_combine_receivable_vals(self, payment_method, amount, amount_converted):
+        partial_vals = {
+            'account_id': payment_method.receivable_account_id.id,
+            'move_id': self.move_id.id,
+            'name': '%s - %s' % (self.name, payment_method.name)
+        }
+        return self._debit_amounts(partial_vals, amount, amount_converted)
+
+    def _get_invoice_receivable_vals(self, account_id, amount, amount_converted):
+        partial_vals = {
+            'account_id': account_id,
+            'move_id': self.move_id.id,
+            'name': 'From invoiced orders'
+        }
+        return self._credit_amounts(partial_vals, amount, amount_converted)
+
+    def _get_sale_vals(self, key, amount, amount_converted):
+        account_id, sign, tax_keys = key
+        tax_ids = [tax[0] for tax in tax_keys]
+        applied_taxes = self.env['account.tax'].browse(tax_ids)
+        name = '' if not applied_taxes else '%s group: %s' % ('Sales' if sign == 1 else 'Refund', ', '.join([tax.name for tax in applied_taxes]))
+        base_tags = applied_taxes\
+            .mapped('invoice_repartition_line_ids' if sign == 1 else 'refund_repartition_line_ids')\
+            .filtered(lambda line: line.repartition_type == 'base')\
+            .tag_ids
+        partial_vals = {
+            'name': name,
+            'account_id': account_id,
+            'move_id': self.move_id.id,
+            'tax_ids': [(6, 0, tax_ids)],
+            'tag_ids': [(6, 0, base_tags.ids)],
+        }
+        return self._credit_amounts(partial_vals, amount, amount_converted)
+
+    def _get_tax_vals(self, key, amount, amount_converted, base_amount):
+        account_id, repartition_line_id, tax_id, tag_ids = key
+        tax = self.env['account.tax'].browse(tax_id)
+        partial_args = {
+            'name': tax.name,
+            'account_id': account_id,
+            'move_id': self.move_id.id,
+            'tax_base_amount': base_amount,
+            'tax_repartition_line_id': repartition_line_id,
+            'tag_ids': [(6, 0, tag_ids)],
+        }
+        return self._credit_amounts(partial_args, amount, amount_converted)
+
+    def _get_stock_expense_vals(self, exp_account, amount, amount_converted):
+        partial_args = {'account_id': exp_account.id, 'move_id': self.move_id.id}
+        return self._debit_amounts(partial_args, amount, amount_converted)
+
+    def _get_stock_output_vals(self, out_account, amount, amount_converted):
+        partial_args = {'account_id': out_account.id, 'move_id': self.move_id.id}
+        return self._credit_amounts(partial_args, amount, amount_converted)
+
+    def _get_statement_line_vals(self, statement, receivable_account, amount, amount_converted):
```

Comment:
amount converted is not used. I think that it is not needed in this case, because all lines will always be in the currency of the statement.

If you need it you should fill the fields 'amount_currency' and 'currency_id' but I think it is only when you have a statement line in a currency different of the statement one (and has no linked on how it's done on the move). And in pos, all lines will be made in the currency of the statement.

Reply:
Precisely. I'm removing the amount_converted in the params list.